src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1286,8 +1286,13 @@ private static bool IsTypeSpecForTypicalInstantiation(TypeDesc t)
                 // we have to apply more restrictive rules
                 // These rules are related to the "inlining rules" as far as the
                 // boundaries of a version bubble are concerned.
+                // This check is different between CG1 and CG2. CG1 considers two types in the same version bubble
+                // if their assemblies are in the same bubble, or if the NonVersionableTypeAttribute is present on the type.
+                // CG2 checks a method cache that it builds with a bunch of new code.
                 if (!_compilation.NodeFactory.CompilationModuleGroup.VersionsWithMethodBody(callerMethod) ||
-                    !_compilation.NodeFactory.CompilationModuleGroup.VersionsWithMethodBody(targetMethod))
+                    // check the Typical TargetMethod, not the Instantiation
+                    !_compilation.NodeFactory.CompilationModuleGroup.VersionsWithMethodBody(targetMethod.IsTypicalMethodDefinition ?
```

Comment:
You unconditionally use `targetMethod.GetTypicalMethodDefinition()` here since if the type is not generic it will just return back the `targetMethod` (https://github.com/dotnet/runtime/blob/01267990a4980352495eae1b230de44858114e0c/src/coreclr/src/tools/Common/TypeSystem/Common/MethodDesc.cs#L538)

Reply:
ah, will fix.